synapseclient/models/services/storable_entity_components.py

```diff
@@ -203,23 +203,24 @@ async def _store_activity_and_annotations(
 
     if (
         hasattr(root_resource, "activity")
-        and (root_resource.activity or last_persistent_instance)
+        and root_resource.activity is not None
         and (
             last_persistent_instance is None
-            or last_persistent_instance.activity != root_resource.activity
+            or (
+                (last_persistent_instance.activity != root_resource.activity)
+                or (
+                    hasattr(root_resource, "associate_activity_to_new_version")
+                    and root_resource.associate_activity_to_new_version
+                    and last_persistent_instance.version_number
+                    != root_resource.version_number
+                )
+            )
         )
     ):
-        if root_resource.activity:
-            result = await root_resource.activity.store_async(
-                parent=root_resource, synapse_client=synapse_client
-            )
-            root_resource.activity = result
-        else:
-            from synapseclient.models import Activity
-
-            await Activity.disassociate_from_entity_async(
```

Comment:
Why do we remove this part that disassociates the activity from the entity if it didn't meet the requirements for it to be stored to synapse?

Reply:
I made a change here that stop the store process from automatically removing an Activity during the store operation. I am currently setting this up such that you need to directly interact with the Activity (`disassociate_from_entity_async` or `delete_async`) to modify these relationships.

I think more discussion needs to happen if we want to automatically modify the activity/provenance relationship for folks like this code was doing.